src/functions/dockerInstead.ts

```diff
@@ -0,0 +1,105 @@
+import readline from "readline";
+import fs from "fs";
+import fsp from "node:fs/promises";
+import path from "path";
+import { fileURLToPath } from "url";
+import { Eta } from "eta";
+import { execSync } from "child_process";
+
+export function isDockerInstalled(): boolean {
+  try {
+    const dockerVer = execSync("docker --version", { stdio: "pipe" }).toString().trim();
+    execSync("docker --version", { stdio: "pipe" });
+    console.log(`Docker is already installed: ${dockerVer}`);
+    return true;
+  } catch (error: unknown) {
+    console.log(error);
+    return false;
+  }
+}
+
+export function noDocker(): string {
+  return "Docker is not installed. Please install Docker (https://www.docker.com) and try again.";
+  process.exit();
+}
+
+export async function wouldPreferDocker(): Promise<boolean> {
+  const rl = readline.createInterface({
+    input: process.stdin,
+    output: process.stdout,
+  });
+
+  const dockerMessage =
+    "Would you prefer to use Docker to run the project? Recommended if you are not comfortable managing your own dependencies. (default: no) ";
+
+  return new Promise((resolve) => {
+    rl.question(dockerMessage, (answer) => {
+      rl.close();
+      const yesOrNo = answer.trim() || "nillion-quickstart";
+      if (/y|yes|n|no/.test(yesOrNo)) {
+        resolve(yesOrNo === "y" || yesOrNo === "yes");
+      } else {
+        console.log("Using default: no Docker container.");
+        resolve(false);
+      }
+    });
```

Comment:
I'd like to replace this part of the code with something more robust like prompt.js or inquirer.js!